view/frontend/web/js/view/payment/method-renderer/creditcards.js

```diff
@@ -25,326 +25,261 @@ define(
+                        fontStyle: "normal",
+                        fontWeight: 400,
+                        fontFamily: 'Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif',
+                        textAlign: 'left',
+                        background: '#fefefe',
+                        color: '#333333',
+                        placeholderColor: '#888888',
+                        borderRadius: '5px',
+                        padding: '8px 10px',
+                        boxShadow: 'none',
+                        transition: 'border-color 0.2s ease, box-shadow 0.2s ease',
+                        border: '1px solid #d6d6d6',
+                        cardLogoStyling: cardLogoStyling
+                    };
 
-                    return text.replace('%s', this.baseCurrencyCode);
-                },
+                    await sdkClient.mountCardHolderName("#cc-name-wrapper", {
+                        id: "ccname",
+                        placeHolder: "John Doe",
+                        labelSelector: "#cc-name-label",
+                        baseStyling: styling
+                    }).then(field => field.focus());
+
+                    await sdkClient.mountCardNumber("#cc-number-wrapper", {
+                        id: "cc",
+                        placeHolder: "555x xxxx xxxx xxxx",
+                        labelSelector: "#cc-number-label",
+                        baseStyling: styling
+                    });
 
-                /**
-                 * Place order.
-                 *
-                 * placeOrderAction has been changed from Magento_Checkout/js/action/place-order to our own version
-                 * (Buckaroo_Magento2/js/action/place-order) to prevent redirect and handle the response.
-                 */
-                placeOrder: function (data, event) {
-                    var self = this,
-                        placeOrder;
+                    await sdkClient.mountCvc("#cc-cvc-wrapper", {
+                        id: "cvc",
+                        placeHolder: "1234",
+                        labelSelector: "#cc-cvc-label",
+                        baseStyling: styling
+                    });
 
-                    if (event) {
-                        event.preventDefault();
-                    }
+                    await sdkClient.mountExpiryDate("#cc-expiry-wrapper", {
+                        id: "expiry",
+                        placeHolder: "MM / YY",
+                        labelSelector: "#cc-expiry-label",
+                        baseStyling: styling
+                    });
 
-                    if (this.validate() && additionalValidators.validate()) {
-                        this.isPlaceOrderActionAllowed(false);
-                        this.encryptCardData().then(function () {
-                            placeOrder = placeOrderAction(self.getData(), self.redirectAfterPlaceOrder, self.messageContainer);
+                    let payButton = document.getElementById("pay");
```

Comment:
why aren't you using a knockout click binding? at leas ad a prefix to he id you use, maybe the id can be used by something else and the event handler can fail/not trigger correctly